src/aaz_dev/swagger/controller/_example_builder.py

```diff
@@ -62,34 +97,104 @@ def mapping(self, example_dict):
+                cmd_operation=self.cmd_operation,
+                arg_var=arg_var,
+                key=param_name,
+                val=value
+            )
+            if item and item.is_top_level:
                 self.example_items.append((item.arg_option, json.dumps(value)))
 
         return self.example_items
 
-    def build(self, var_prefix, example_dict):
+    def build(self, var_prefix, example_obj, disc=None):
         example_items = []
-        if isinstance(example_dict, list):
+        if isinstance(example_obj, list):
             arg_var = f"{var_prefix}[]"
-            for item in example_dict:
-                example_items += self.build(arg_var, item)
-        elif isinstance(example_dict, dict):
-            for name, value in example_dict.copy().items():
-                item = ExampleItem(command=self.command, arg_var=f"{var_prefix}{{}}.{name}", key=name, val=value)
-                if item.arg is None:
-                    item = ExampleItem(command=self.command, arg_var=f"{var_prefix}.{name}", key=name, val=value)
-
-                example_items += self.build(item.arg_var, value)
-
-                if item.is_top_level:
-                    example_items.append((item.arg_option, json.dumps(value)))
-                elif item.is_flatten:
-                    example_dict.pop(item.key)
-                    for k, v in item.val.items():
-                        example_dict[k] = v
-                elif item.arg_option:
-                    example_dict.pop(item.key)
-                    example_dict[item.arg_option] = item.val
+            item = ExampleItem.new_instance(
+                command=self.command,
+                cmd_operation=self.cmd_operation,
+                arg_var=arg_var
+            )
+            if item:
+                discs = item.discriminators
+                for obj in example_obj:
+                    for disc in discs:
+                        if disc.property not in obj or obj[disc.property] != disc.value:
+                            continue
+
+                        example_items += self.build(arg_var, obj, disc)
+                        break
+                    else:
+                        example_items += self.build(arg_var, obj)
+
+        elif isinstance(example_obj, dict):
+            disc_name = None
+            if disc is not None:  # handle discriminator
+                example_obj.pop(disc.property)  # ignore discriminator prop
+
+                safe_value = self.get_safe_value(disc.value)
+                disc_item = ExampleItem.new_instance(
+                    command=self.command,
+                    cmd_operation=self.cmd_operation,
+                    arg_var=f"{var_prefix}.{safe_value}"
+                )
+
+                if disc_item and (disc_name := disc_item.arg_option):
+                    example_obj[disc_name] = example_obj.copy()
+                    example_items += self.build(disc_item.arg_var, example_obj[disc_name])
```

Comment:
Can this work? Why need to copy and assign in example_obj?
```suggestion
                    example_items += self.build(disc_item.arg_var, example_obj.copy())
```

Reply:
aim to continently trim the inner/outer layer, although copy is OK, deepcopy might be better